Source/WebCore/platform/graphics/Damage.h

```diff
@@ -161,26 +167,52 @@ class Damage {
             add(rect);
     }
 
+    void removeOverlapsWhileClippingToGrid()
```

Comment:
Since this is always called before iterating the rects for painting, I would make this a getter instead of modifying the damage. For non-rectangles case I will always copy, but it's always just one rect.